crypto/ec/curve448/arch_arm_32/f_impl.c

```diff
@@ -0,0 +1,827 @@
+/*
+ * Copyright 2017 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2014 Cryptography Research, Inc.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ *
+ * Originally written by Mike Hamburg
+ */
+
+#include "f_field.h"
+
+static inline void __attribute__((gnu_inline,always_inline))
+smlal (
+    uint64_t *acc,
+    const uint32_t a,
+    const uint32_t b
+) {
+
+#ifdef  __ARMEL__
+    uint32_t lo = *acc, hi = (*acc)>>32;
+    
+    __asm__ __volatile__ ("smlal %[lo], %[hi], %[a], %[b]"
+        : [lo]"+&r"(lo), [hi]"+&r"(hi)
+        : [a]"r"(a), [b]"r"(b));
```

Comment:
I'm going to guess that all this assembler is going to need to get converted to perlasm so that it's works over all our platforms. I'm wondering if we should start with a plain C implementation and add all the assembler implementations at some later time.

Reply:
As I mentioned in the intro to this PR I've only integrated the 32-bit reference implementation which is C only. The platform specific optimisations are still there, but aren't integrated into the build at this point. This code belongs to one of those non-integrated files.